tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,204 @@
+
+  Args:
+    file_pattern: List of paths or patterns of input tfrecord files.
+    tf_transform_output: A TFTransformOutput.
+    batch_size: representing the number of consecutive elements of returned
+      dataset to combine in a single batch
+
+  Returns:
+    A dataset that contains (features, indices) tuple where features is a
+      dictionary of Tensors, and indices is a single Tensor of label indices.
+  """
+  transformed_feature_spec = (
+      tf_transform_output.transformed_feature_spec().copy())
+
+  dataset = tf.data.experimental.make_batched_features_dataset(
+      file_pattern=file_pattern,
+      batch_size=batch_size,
+      features=transformed_feature_spec,
+      reader=_gzip_reader_fn,
+      label_key=transformed_name(LABEL_KEY))
+
+  return dataset
+
+def _freeze_model_by_percentage(model: tf.keras.Model,
+                                percentage: float):
+  """Freeze part of the model based on specified percentage
+
+  Args:
+    model: The keras model need to be partially frozen
+    percentage: the percentage of layers to freeze
+  """
+  percentage = max(0.0, percentage)
+  percentage = min(1.0, percentage)
+  num_layers = len(model.layers)
+  num_layers_to_freeze = int(num_layers * percentage)
+  for idx, layer in enumerate(model.layers):
+    if idx >= num_layers_to_freeze:
+      break
+    layer.trainable = False
+
+def _build_keras_model() -> tf.keras.Model:
+  """Creates a MobileNet model pretrained on ImageNet for finetuning
+
+  Returns:
+    A Keras Model.
+  """
+  tf.keras.backend.set_image_data_format('channels_last')
+  base_model = tf.keras.applications.MobileNet(
+      input_shape=(224, 224, 3), include_top=False, weights='imagenet',
+      pooling='avg')
+
+  _freeze_model_by_percentage(base_model, 0.9)
+
+  model = tf.keras.Sequential([
+      tf.keras.layers.InputLayer(
+          input_shape=(224, 224, 3), name=transformed_name(IMAGE_KEY)),
+      base_model,
+      tf.keras.layers.Dropout(0.3),
+      tf.keras.layers.Dense(10, activation='softmax')
+  ])
+
+  model.compile(
+      loss='sparse_categorical_crossentropy',
+      optimizer=tf.keras.optimizers.RMSprop(lr=0.0015),
+      metrics=['sparse_categorical_accuracy'])
+  model.summary(print_fn=absl.logging.info)
+  return model
+
+# TFX Transform will call this function.
+def preprocessing_fn(inputs):
+  """tf.transform's callback function for preprocessing inputs.
+
+  Args:
+    inputs: map from feature keys to raw not-yet-transformed features.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
+  """
+  outputs = {}
+
+  image_features = tf.map_fn(lambda x: tf.io.decode_png(x[0], channels=3), inputs[IMAGE_KEY], dtype=tf.uint8)
+  image_features = tf.cast(image_features, tf.float32)
+
+  # The MobileNet we use was trained on ImageNet, which has image size 224 x 224.
+  # We resize CIFAR10 images to match that size
+  image_features = tf.image.resize(image_features, [224, 224])
```

Comment:
There are multiple ways of resizing. Why do we believe this provides the best quality for our dataset?

Reply:
See https://www.tensorflow.org/api_docs/python/tf/image/ResizeMethod for options.

Reply:
hmm I think the default option (bilinear pooling) is the most common practice. It worked well as the model got ~90% acc on validation set.

Reply:
lets push to see how high we can go. Just because we get ~90% accuracy on an easy dataset doesn't mean that we can't push for more.